### DISCUSSION TITLE
Is the following code snippet legit in Astro?

### DISCUSSION THREAD

**Discussion by Ali-Hussein-dev** [withastro/starlight#2069] [Q&A] [answered]
I am not an Astro dev, I need to add a script to `mdx` file, but it is not working in prod, what is the right way to do it?

```mdx

{/* src/content/docs/index.mdx */}

{!import.meta.env.DEV && (
<head>
  <script
    defer
    src="..."
    data-website-id="...."
  ></script>
</head>
)} 
```

**Comment by HiDeoo** [comment]
👋 Do you need the script to be added only on the homepage or to all pages?

**Reply by Ali-Hussein-dev** [reply]
In Nextjs I added to the root `layout.tsx`, so I suppose all pages

**Reply by HiDeoo** [reply, marked as answer]
> In Nextjs I added to the root `layout.tsx`, so I suppose all pages

In that case, you should be able to use the [`head`](https://starlight.astro.build/reference/configuration/#head) global configuration option inside the `astro.config.mjs` configuration file to add your script to the head of every page.

The example in the documentation should be relative close to what you shared:

```js
starlight({
  head: [
    {
      tag: "script",
      attrs: {
        src: "…",
        "data-website-id": "MY-ID",
        defer: true,
      },
    },
  ],
});
```

**Reply by Ali-Hussein-dev** [reply]
> > In Nextjs I added to the root `layout.tsx`, so I suppose all pages
> 
> In that case, you should be able to use the [`head`](https://starlight.astro.build/reference/configuration/#head) global configuration option inside the `astro.config.mjs` configuration file to add your script to the head of every page.
> 
> The example in the documentation should be relative close to what you shared:
> 
> ```js
> starlight({
>   head: [
>     {
>       tag: "script",
>       attrs: {
>         src: "…",
>         "data-website-id": "MY-ID",
>         defer: true,
>       },
>     },
>   ],
> });
> ```

and if I want to add only to the root page, what would look like?
the integration doesn't work as expected if the script renders on every page. I need to try on the root page only.

**Reply by HiDeoo** [reply]
> and if I want to add only to the root page, what would look like?

If you just want to target a specific page, you can use the equivalent [`head` frontmatter field](https://starlight.astro.build/reference/frontmatter/#head) which should look like:

```mdx
---
# src/content/docs/index.mdx
title: Homepage
head:
  - tag: script
    attrs:
      src: '…'
      data-website-id: 'MY-ID'
      defer: true
---
```
